test/service.js

```diff
@@ -380,93 +381,98 @@ describe('Service', function() {
           },
           TargetHealth: {
-            State: "unhealthy"
-          }
-        }
+            State: 'unhealthy',
+          },
+        },
       ]);
 
       setServiceDependencyFixture('tasks', [
         {
           taskArn: 'arn:aws:ecs:us-east-1:12345789012:task/mycluster/abcdefgh',
           containers: [
             {
-                "name": "app"
-            }
+              'name': 'app',
+            },
           ],
           attachments: [
             {
               'type': 'ElasticNetworkInterface',
               'details': [
                 {
                   'name': 'privateIPv4Address',
-                  'value': '192.0.2.1'
+                  'value': '192.0.2.1',
                 },
-              ]
-            }
-          ]
+              ],
+            },
+          ],
         },
         {
           taskArn: 'arn:aws:ecs:us-east-1:12345789012:task/mycluster/bcdefghi',
           containers: [
             {
-                "name": "app"
-            }
+              'name': 'app',
+            },
           ],
           attachments: [
             {
               'type': 'ElasticNetworkInterface',
               'details': [
                 {
                   'name': 'privateIPv4Address',
-                  'value': '192.0.2.2'
+                  'value': '192.0.2.2',
                 },
-              ]
-            }
-          ]
+              ],
+            },
+          ],
         },
         {
           taskArn: 'arn:aws:ecs:us-east-1:12345789012:task/mycluster/cdefghij',
           containers: [
             {
-                "name": "app"
-            }
+              'name': 'app',
+            },
           ],
           attachments: [
             {
               'type': 'ElasticNetworkInterface',
               'details': [
                 {
                   'name': 'privateIPv4Address',
-                  'value': '192.0.2.3'
+                  'value': '192.0.2.3',
                 },
-              ]
-            }
-          ]
+              ],
+            },
+          ],
         },
       ]);
     });
 
     it('should report task health', function(done) {
-      AWS.mock('ECS', 'describeServices', function (params, cb){
+      AWS.mock('ECS', 'describeServices', function(params, cb) {
         cb(null, fixtures['fargateDeployment']);
       });
+      // lib/resources/tasks has an ECS/describeTasks call that needs hooking
+      // to prevent the actual SDK from trying to make a call. We don't see the
+      // results from it here so no-op is fine.
+      AWS.mock('ECS', 'describeTasks', () => {});
 
-      var service = new Service({clusterArn: 'arn:aws:ecs:us-east-1:12345789012:cluster/mycluster', serviceName: 'my-test-application'});
+      const service = new Service({clusterArn: 'arn:aws:ecs:us-east-1:12345789012:cluster/mycluster', serviceName: 'my-test-application'});
 
       service.on('updated', function() {
+        service.destroy();
```

Comment:
should this be here?

Reply:
Actually this does make sense, `destroy()` just has a misleading name. We should probably call this first in all of the test sections where we are expecting no further state changes.